packages/patternfly-4/react-core/src/experimental/components/DataToolbar/DataToolbarItemWithChipGroup.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import { DataToolbarItem, DataToolbarItemProps } from './DataToolbarItem';
+import { ChipGroup, Chip, ChipGroupToolbarItem } from '../../../components/ChipGroup';
+import { DataToolbarContext } from './DataToolbarUtils';
+import styles from '@patternfly/react-styles/css/components/DataToolbar/data-toolbar';
+import { getModifier } from '@patternfly/react-styles';
+
+export interface DataToolbarItemWithChipGroupProps extends DataToolbarItemProps {
+  /** An array of strings to be displayed as chips in the expandable content */
+  chips?: string[];
+  /** Callback passed by consumer used to delete a chip from the chips[] */
+  deleteChip?: (chip: string) => void;
+  /** Content to be rendered inside the Data toolbar item associated with the chip group */
+  children: React.ReactNode;
+  /** Unique label for the chip group */
+  label: string;
+}
+
+interface DataToolbarItemWithChipGroupState {
+  isMounted: boolean;
+}
+
+export class DataToolbarItemWithChipGroup
+  extends React.Component<DataToolbarItemWithChipGroupProps, DataToolbarItemWithChipGroupState> {
+
+  static defaultProps = {
+    chips: [] as string[]
+  };
+
+  constructor(props: DataToolbarItemWithChipGroupProps) {
+    super(props);
+    this.state = {
+      isMounted: false
+    };
+  }
+
+  componentDidMount() {
+    /* eslint-disable-next-line */
+    this.setState({ isMounted: true });
+  }
+
+  render() {
+    const { children, chips, deleteChip, label, ...props } = this.props;
+
+    return (
+      <DataToolbarContext.Consumer>
+        {({ isExpanded, chipGroupContentRef, updateShowClearFiltersButton }) => {
+
+          if (chips.length > 0) {
+            updateShowClearFiltersButton(true);
```

Comment:
Updating state at render time is an antipattern we should avoid (it can lead to render loops and incorrect state, since React doesn't guarantee a render on every update). If you wanted to implement what this code is trying to do (call `updateShowClearFiltersButton` when the value of `chips` changes), that should be done in `componentDidUpdate` instead of `render`.

However, we're duplicating sources of truth here; ideally, `showClearFiltersButton` shouldn't be its own value in state at all, it should be derived from whether or not there are any filters applied across all chip groups.
Duplicating that state is problematic, I think it's causing the bug @karelhala noticed:
> if you remove all filters trough their x buttons the Clear all filters is still present.

This call (`if (chips.length > 0) { updateShowClearFiltersButton(true); }`) makes sure that when any chips appear the Clear all filters button is shown, but there is nothing making sure that when the last chip disappears the Clear all filters button is hidden. And since the DataToolbarItemWithChipGroup components only know about their own chips, there's no way for them to know that a chip being deleted is the last chip in the whole toolbar, so they can't confidently call that method with `false`.

I think the real way to fix all these problems is just to have `showClearFiltersButton` be a prop of `DataToolbar`, instead of having it in state, and to get rid of the whole `updateShowClearFiltersButton` method entirely. The example's root component is the only one who knows whether there are no chips left in any of the DataToolbarItems, so that prop in the example would just be defined as `showClearFiltersButton={statusSelection.length > 0 && riskSelection.length > 0}`.

The only true way to avoid that prop and make this truly "automatic" would be for the parent DataToolbar to "know about" the `chips` props of its children, which would mean they call some method to "register and unregister" their chips when they update... but that sounds like a huge mess, and it violates the React principle of one-way data flow. I think the right solution is to just make it a prop.